lib/playlist.rb

```diff
@@ -0,0 +1,42 @@
+class Playlist < ActiveRecord::Base
+  validates_presence_of :created_at
+
+  has_many :playlist_songs
+  has_many :songs, through: :playlist_songs
+
+  def alphabetized_hash
+    songs.group_by{ |s| s.alpha_categorize }
+  end
+
+  def alphabet_winners_hash
+    Hash[self.alphabetized_hash.map { |k,a| [k,a.max_by {|song| song.total_votes_current_week }]}]
```

Comment:
One-liners are great, but I think ending in `}]}]` is past my personal complexity threshold. I'm fine with this being a little longer if it makes it any easier to read.